### PR TITLE
Catalog of my things

### PR DESCRIPTION
**Pull request by ciraganenicole**
### Here is the link to the video [Ruby-capstone](https://drive.google.com/file/d/13i_VC3Ud1sR-zEDJhIF21BVEAx5e_nj7/view?usp=sharing)

For this project we:
 
#### 1. Implemented the following features:

- List all books
- List all music albums
- List of games
- List all genres
- List all labels 
- List all authors
- List all sources
- Add a book
- Add a music album
- Add a game

#### 3. Create a schema.sql file with tables that will be analogical to the structure of the classes in your app

### CODE REVIEW

File: model/item.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @genre = genre
+  end
+
+  def set_author(author:)
+    @author = author
+  end
+
+  def set_label(label:)
+    @label = label
+  end
```
Comment (mikethreels) [inline model/item.rb:22]: - [ ] the one to many relationship should include the following line.
```
genre.items.push(self) unless genre.items.include?(self)
```
the same goes for Author and Label

File: model/book.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  def can_be_archived?
+    super || cover_state == 'bad'
+  end
+end
+
+# l = Book.new(cover_state: 'bad', publisher: nil)
+# # puts Date.today.to_time.year - l.publish_date.to_time.year
+# puts l.move_to_archive
+# puts l.archived
```
Comment (mikethreels) [inline model/book.rb:30]: - [ ] please remove all commented-out code it will make your codebase look unorganised and harder to read.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢🟢🟢 

Hi Team,

Good Job working on the changes asked by the previous reviewer 👍 

MaxSinclairGIF

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights

✔️ Correct use of Gitflow
✔️ No linter error
✔️ Good commit messages
✔️ App works fine
✔️ Good presentation 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**